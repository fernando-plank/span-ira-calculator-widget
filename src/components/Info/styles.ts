import styled from 'styled-components'

export const Tooltip = styled.div`
  width: max-content;
  display: contents;

  svg {
    margin: 0px 10px;
  }
`

export const InfoContent = styled.h4`
  background-color: ${({ theme }) => theme.colors.gray};
  width: max-content;
  padding: 12px 16px;
  border-radius: 5px;
  position: absolute;
  color: ${({ theme }) => theme.colors.black};
  max-width: 300px;
  text-transform: none;
  font-style: normal;
  font-weight: 500;
  font-size: 16px;
  line-height: 24px;
`
