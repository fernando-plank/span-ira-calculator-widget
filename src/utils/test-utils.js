var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
/* eslint-disable import/export */
import React from 'react';
import theme from '@styles/theme';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
jest.mock('../components/Icon', () => {
    return jest.fn(() => React.createElement("div", null, "mocked icon"));
});
const customRender = (ui, _a = {}) => {
    var renderOptions = __rest(_a, []);
    return render(React.createElement(ThemeProvider, { theme: theme }, ui), renderOptions);
};
export * from '@testing-library/react';
export { customRender as render };
