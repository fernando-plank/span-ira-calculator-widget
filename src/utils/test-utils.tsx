/* eslint-disable import/export */
import { ReactElement } from 'react'
import { RenderOptions, render } from '@testing-library/react'

import theme from '@styles/theme'
import { ThemeProvider } from 'styled-components'

jest.mock('../components/Icon', () => {
  return jest.fn(() => <div>mocked icon</div>)
})

const customRender = (ui: ReactElement, { ...options }: RenderOptions = {}) =>
  render(<ThemeProvider theme={theme}>{ui}</ThemeProvider>, options)

export * from '@testing-library/react'
export { customRender as render }
