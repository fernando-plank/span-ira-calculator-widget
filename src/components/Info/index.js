import React from 'react';
import { useState } from 'react';
import Icon from '@components/Icon';
import * as S from './styles';
const Info = (props) => {
    const [showContent, setShowContent] = useState(false);
    return (React.createElement(S.Tooltip, { "data-testid": "tooltip", onMouseEnter: () => setShowContent(true), onMouseLeave: () => setShowContent(false) },
        showContent && (React.createElement(S.InfoContent, { "data-testid": "tooltip-content" }, props.text)),
        React.createElement(Icon, { "data-testid": "tooltip-data", style: { margin: '0px 5px' }, icon: 'Info' })));
};
export default Info;
