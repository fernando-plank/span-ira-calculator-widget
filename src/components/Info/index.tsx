import React, { ReactChild } from 'react'
import { useState } from 'react'

import Icon from '@components/Icon'

import * as S from './styles'

const Info = (props: { text: ReactChild }) => {
  const [showContent, setShowContent] = useState(false)
  return (
    <S.Tooltip
      data-testid="tooltip"
      onMouseEnter={() => setShowContent(true)}
      onMouseLeave={() => setShowContent(false)}
    >
      {showContent && (
        <S.InfoContent data-testid="tooltip-content">
          {props.text}
        </S.InfoContent>
      )}
      <Icon data-testid="tooltip-data" icon={'Info'} />
    </S.Tooltip>
  )
}
export default Info
