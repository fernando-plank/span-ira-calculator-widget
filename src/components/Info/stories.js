import React from 'react';
import Info from '.';
export default {
    title: 'Info',
    component: Info
};
export const Default = () => React.createElement(Info, { text: 'Info content' });
