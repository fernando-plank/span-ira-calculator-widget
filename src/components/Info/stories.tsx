import React from 'react'
import { Story, Meta } from '@storybook/react'

import Info from '.'

export default {
  title: 'Info',
  component: Info
} as Meta

export const Default: Story = () => <Info text={'Info content'} />
