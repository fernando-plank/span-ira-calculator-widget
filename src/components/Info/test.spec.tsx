import { fireEvent, screen } from '@testing-library/react'
import { render } from '@utils/test-utils'

import Info from '.'

jest.mock('../Icon', () => {
  return jest.fn(() => <div>mocked icon</div>)
})

describe('<Info />', () => {
  it('should show tooltip when click on icon', () => {
    const { container } = render(<Info text="Info Content" />)

    expect(screen.getByTestId('tooltip')).toBeInTheDocument()

    fireEvent.mouseEnter(screen.getByTestId('tooltip'))

    expect(screen.getByText('Info Content')).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should show tooltip when click on icon', () => {
    const { container } = render(<Info text="Info Content" />)

    expect(screen.getByTestId('tooltip')).toBeInTheDocument()

    fireEvent.mouseEnter(screen.getByTestId('tooltip'))
    fireEvent.mouseLeave(screen.getByTestId('tooltip'))

    expect(screen.queryByText('Info Content')).not.toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
