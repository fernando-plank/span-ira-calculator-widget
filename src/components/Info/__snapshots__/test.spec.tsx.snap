// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Info /> should show tooltip when click on icon 1`] = `
<div
  class="sc-bgqQcB hxTPvd"
  data-testid="tooltip"
>
  <h4
    class="sc-gTRrQi daYTAa"
    data-testid="tooltip-content"
  >
    Info Content
  </h4>
  <div>
    mocked icon
  </div>
</div>
`;

exports[`<Info /> should show tooltip when click on icon 2`] = `
<div
  class="sc-bgqQcB hxTPvd"
  data-testid="tooltip"
>
  <div>
    mocked icon
  </div>
</div>
`;
